src/idl_gen_dlang.cpp

```diff
@@ -0,0 +1,690 @@
+/*
+ * Copyright 2014 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// independent from idl_parser, since this code is not needed for most clients
+
+#include <list>
+#include <algorithm>
+
+#include "flatbuffers/flatbuffers.h"
+#include "flatbuffers/idl.h"
+#include "flatbuffers/util.h"
+
+namespace flatbuffers {
+
+namespace dlang {
+
+static std::string GenGetter(const Type &type);
+static std::string GenMethod( const Type &type);
+
+static std::string GenTypeBasic(const Type &type);
+static std::string GenTypeGet(const Type &type);
+
+static std::string FunctionStart(char upper) {
```

Comment:
You copied this from Java/C#, where this function switches between upper/lower case. In D, this function is constant. So please remove this function, and put the actual character in place for the callers.